queries.sql

```diff
@@ -15,3 +15,33 @@ SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+/* Query and update table */
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT DELETE_DOB;
+UPDATE animals SET weight_kg = (weight_kg * -1);
+ROLLBACK TO DELETE_DOB;
+UPDATE animals SET weight_kg = (weight_kg * -1) WHERE weight_kg < 0; 
+COMMIT TRANSACTION;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
+SELECT neutered , MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY neutered;
```

Comment:
* I think you can use the species column added to the animal table to write our queries 
* For example for this question ( What is the minimum and maximum weight of each type of animal? ) answer (
` SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
   GROUP BY species;`  here we use species for differentiating animal types 
``` )

Reply:
Used species to differentiate animal types